pkgs/os-specific/linux/iwd/0001-agent-network_connect_psk-check-if-the-config-file-f.patch

```diff
@@ -0,0 +1,60 @@
+From c16e0c347c0105b5665747a95989907ab64fb0fb Mon Sep 17 00:00:00 2001
+From: Maximilian Bosch <maximilian@mbosch.me>
+Date: Mon, 10 Feb 2020 21:14:06 +0100
+Subject: [PATCH] agent/network_connect_psk: check if the config-file for the
+ SSID points to a store path
+
+NixOS-specific patch. When declaratively configuring networks, `iwd`
+shouldn't modify their config files.
+---
+ src/dbus.c    |  6 ++++++
+ src/network.c | 15 ++++++++++++++-
+ 2 files changed, 20 insertions(+), 1 deletion(-)
+
+diff --git a/src/dbus.c b/src/dbus.c
+index ceede5c..5bdd04b 100644
+--- a/src/dbus.c
++++ b/src/dbus.c
+@@ -114,6 +114,12 @@ struct l_dbus_message *dbus_error_not_supported(struct l_dbus_message *msg)
+ 					"Operation not supported");
+ }
+ 
++struct l_dbus_message *dbus_error_file_is_a_nix_store_Path(struct l_dbus_message *msg, char *ssid)
++{
++	return l_dbus_message_new_error(msg, IWD_SERVICE ".NotSupportNixOS",
++					"Cannot update SSID %s as it's configured in a file that points to a store path! Please update your NixOS config accordingly!", ssid);
++}
++
+ struct l_dbus_message *dbus_error_no_agent(struct l_dbus_message *msg)
+ {
+ 	return l_dbus_message_new_error(msg, IWD_SERVICE ".NoAgent",
+diff --git a/src/network.c b/src/network.c
+index 29cd4e1..5010003 100644
+--- a/src/network.c
++++ b/src/network.c
+@@ -843,8 +843,21 @@ static struct l_dbus_message *network_connect_psk(struct network *network,
+ 
+ 	l_debug("ask_passphrase: %s",
+ 		network->ask_passphrase ? "true" : "false");
+-
+ 	if (network->ask_passphrase) {
++		char *store_dir = getenv("NIX_STORE_DIR");
```

Comment:
Perhaps this should be taken from NixOS config values instead of runtime environment?

Reply:
Is there any config option for this I'm missing?

Reply:
builtins.storeDir

Reply:
Fixed now.